bugbug/generative_model_tool.py

```diff
@@ -3,40 +3,76 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import re
 from abc import ABC, abstractmethod
 from logging import INFO, basicConfig, getLogger
 from typing import Any
 
-from langchain_community.llms import HumanInputLLM
-from langchain_openai import AzureChatOpenAI, ChatOpenAI
-
 from bugbug.utils import get_secret
 
 basicConfig(level=INFO)
 logger = getLogger(__name__)
 
 
+def create_human_llm():
+    from langchain_community.llms import HumanInputLLM
+
+    return HumanInputLLM()
+
+
+def create_openai_llm():
+    from langchain_openai import ChatOpenAI
+
+    return ChatOpenAI(
+        model_name="gpt-4o-2024-05-13",
+        api_key=get_secret("OPENAI_API_KEY"),
+        temperature=0.2,
+    )
+
+
+def create_azureopenai_llm():
+    from langchain_openai import AzureChatOpenAI
+
+    return AzureChatOpenAI(
+        azure_endpoint=get_secret("OPENAI_API_ENDPOINT"),
+        azure_deployment=get_secret("OPENAI_API_DEPLOY"),
+        api_key=get_secret("OPENAI_API_KEY"),
+        api_version=get_secret("OPENAI_API_VERSION"),
+        temperature=0.2,
+    )
+
+
+def create_anthropic_llm():
+    from langchain_anthropic import ChatAnthropic
+
+    return ChatAnthropic(
+        model_name="claude-3-5-sonnet-20240620",
+        api_key=get_secret("ANTHROPIC_API_KEY"),
+    )
+
+
+def create_mistral_llm():
+    from langchain_mistralai import ChatMistralAI
+
+    return ChatMistralAI(
+        model_name="mistral-large-latest",
+        api_key=get_secret("MISTRAL_API_KEY"),
+    )
+
+
+AVAILABLE_LLMS = [
+    match.group(1)
+    for name in list(globals())
+    if (match := re.search(r"create_(.*?)_llm", name))
+]
+
+
 def create_llm(llm):
-    openai_temperature = 0.2
-    if llm == "human":
-        return HumanInputLLM()
-    elif llm == "openai":
-        return ChatOpenAI(
-            model_name="gpt-4o-2024-05-13",
-            api_key=get_secret("OPENAI_API_KEY"),
-            temperature=openai_temperature,
-        )
-    elif llm == "azureopenai":
-        return AzureChatOpenAI(
-            azure_endpoint=get_secret("OPENAI_API_ENDPOINT"),
-            azure_deployment=get_secret("OPENAI_API_DEPLOY"),
-            api_key=get_secret("OPENAI_API_KEY"),
-            api_version=get_secret("OPENAI_API_VERSION"),
-            temperature=openai_temperature,
-        )
-    else:
+    if llm not in AVAILABLE_LLMS:
         raise NotImplementedError
 
+    return globals()[f"create_{llm}_llm"]()
```

Comment:
I did it to avoid having to add both a new function and a new if-return statement whenever adding a new model. This way one just needs to add a new function

Reply:
Yes, I noticed that. It is a trade-off between convenience and clarity. It is worth it if people will be frequently trying different stuff.